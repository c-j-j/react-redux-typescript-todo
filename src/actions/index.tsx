import * as constants from '../constants';

export interface AddTodo {
  type: constants.ADD_TODO;
  text: string;
}

export interface RemoveTodo {
  type: constants.REMOVE_TODO;
}

export type TodoAction = AddTodo | RemoveTodo;

export function addTodo(text: string): AddTodo {
  return {
    type: constants.ADD_TODO,
    text
  };
}

export function removeTodo(): RemoveTodo {
  return {
    type: constants.REMOVE_TODO
  };
}