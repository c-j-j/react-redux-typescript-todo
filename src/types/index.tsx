export interface StoreState {
  items: string[];
}