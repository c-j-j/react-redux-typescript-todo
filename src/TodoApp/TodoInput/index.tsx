import { connect, Dispatch } from 'react-redux';
import { Presentation, Props } from './presentation';
import * as actions from '../../actions';

function mapDispatchToProps(dispatch: Dispatch<actions.AddTodo>): Props {
  return {
    addTodo: (text: string) => dispatch(actions.addTodo(text))
  };
}

export default connect(null, mapDispatchToProps)(Presentation);