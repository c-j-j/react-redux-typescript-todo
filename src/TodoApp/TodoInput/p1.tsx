import * as React from 'react';
import { ChangeEvent, FormEvent } from 'react';
import { AddTodo } from '../../actions';

export interface Props {
  addTodo: (text: string) => AddTodo;
}

interface State {
  currentItem: string;
}

export class Presentation extends React.Component<Props, State> {
  componentWillMount() {
    this.setState({
      currentItem: ''
    });
  }

  addTodoItem(event: FormEvent<HTMLFormElement>) {
    event.preventDefault();
    this.props.addTodo(this.state.currentItem);
  }

  appendToCurrentItem(event: ChangeEvent<HTMLInputElement>) {
    this.setState({currentItem: event.target.value});
  }

  render() {
    return (
      <form onSubmit={(e) => this.addTodoItem(e)}>
        <input type="text" placeholder="Enter todo item" onChange={(e) => this.appendToCurrentItem(e)} />
      </form>
    );
  }
}