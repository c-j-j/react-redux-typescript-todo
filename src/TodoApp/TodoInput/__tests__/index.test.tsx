import * as React from 'react';
import { shallow } from 'enzyme';
import TodoInput from '../index';
import configureMockStore, { MockStore } from 'redux-mock-store';
import { StoreState } from '../../../types';
import { tryAddTodo } from '../../../actions';

describe('TodoInput Container Test', () => {
  let store: MockStore<StoreState>;

  beforeEach(() => {
    store = configureMockStore<StoreState>()();
  });

  it('adds an item by invoking the prop passed to presentation', () => {
    const wrapper = shallow(<TodoInput />, { context: { store }});

    wrapper.prop('addTodo')('item');
    const actions = store.getActions();

    expect(actions[0]).toEqual(tryAddTodo('item'));
  });

  it('adds an item to actions when form is submitted', () => {
    const wrapper = shallow(<TodoInput />, { context: { store }});
    const form = wrapper.dive();
    form.find('form > input').simulate('change', {target: {value: 'item'}});
    form.find('form').simulate('submit', {preventDefault: () => {}});
    const actions = store.getActions();

    expect(actions[0]).toEqual(tryAddTodo('item'));
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<TodoInput />, { context: { store }});

    expect(wrapper.dive().html()).toMatchSnapshot();
  });
});