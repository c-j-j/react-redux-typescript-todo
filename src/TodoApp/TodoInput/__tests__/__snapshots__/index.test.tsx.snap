// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput Container Test matches snapshot 1`] = `"<form><input type=\\"text\\" placeholder=\\"Enter todo item\\"/></form>"`;
